docker-images/kafka/scripts/kafka_run.sh

```diff
@@ -47,6 +43,26 @@ echo "Starting Kafka with configuration:"
 ./kafka_config_generator.sh | tee /tmp/strimzi.properties | sed -e 's/sasl.jaas.config=.*/sasl.jaas.config=[hidden]/g' -e 's/password=.*/password=[hidden]/g'
 echo ""
 
+# Prepare for Kraft
+if [ -n "$CLUSTER_ID" ]; then
+  KRAFT_LOG_DIR=$(grep "log\.dirs=" /tmp/strimzi.properties | sed "s/log\.dirs=*//")
+
+  if [ ! -f "$KRAFT_LOG_DIR/meta.properties" ]; then
+    echo "Formatting Kraft storage"
+    mkdir -p "$KRAFT_LOG_DIR"
+    ./bin/kafka-storage.sh format -t "$CLUSTER_ID" -c /tmp/strimzi.properties
+  else
+    echo "Kraft storage is already formatted"
+  fi
```

Comment:
Should the trigger for this be the non-existence of `metadata.properties`. Maybe it should be done in an initcontainer?